File: Hunt_Career_Cypress/cypress/fixtures/registerData.js
import { generateEmail, generateRandomPassword, generateRandomPhoneNumber, getRandomFirstName, getRandomLastName } from "../utils/utils";

export const registerData = {
    validUser: (() => {
        const password = generateRandomPassword();
        return {
            firstName: getRandomFirstName(),
            lastName: getRandomLastName(),
            email: generateEmail('normal'),
            password: password,
            confirmPassword: password,
            phoneNumber: generateRandomPhoneNumber()
        };
    })(),

    duplicateUser: {
        firstName: "Pavan", 
        lastName: "U",
        email: "upavan@mailinator.com",
        password: "Test@1234",
        confirmPassword: "Test@1234",
        phoneNumber: "1234567890"
    },

    noFirstName: (() => {
        const password = generateRandomPassword();
        return {
            firstName: "",
            lastName: getRandomLastName(),
            email: generateEmail('normal'),
            password: password,
            confirmPassword: password,
            phoneNumber: generateRandomPhoneNumber()
        };
    })(),

    noLastName: (() => {
        const password = generateRandomPassword();
        return {
            firstName: getRandomFirstName(),
            lastName: "",
            email: generateEmail('normal'),
            password: password,
            confirmPassword: password,
            phoneNumber: generateRandomPhoneNumber()
        };
    })(),

    noEmailName: (() => {
        const password = generateRandomPassword();
        return {
            firstName: getRandomFirstName(),
            lastName: getRandomLastName(),
            email: "",
            password: password,
            confirmPassword: password,
            phoneNumber: generateRandomPhoneNumber()
        };
    })(),

    noPasswordName: (() => {
        const password = generateRandomPassword();
        return {
            firstName: getRandomFirstName(),
            lastName: getRandomLastName(),
            email: generateEmail('normal'),
            password: "",
            confirmPassword: password,
            phoneNumber: generateRandomPhoneNumber()
        };
    })(),

    passwordWithLessCharacters: {
        firstName: getRandomFirstName(),
        lastName: getRandomLastName(),
        email: generateEmail('normal'),
        password: "Test",
        confirmPassword: "Test",
        phoneNumber: generateRandomPhoneNumber()
    },

    noConfirmPasswordName: (() => {
        const password = generateRandomPassword();
        return {
            firstName: getRandomFirstName(),
            lastName: getRandomLastName(),
            email: generateEmail('normal'),
            password: password,
            confirmPassword: "",  // confirm password intentionally empty
            phoneNumber: generateRandomPhoneNumber()
        };
    })(),

    noPhoneNumberName: (() => {
        const password = generateRandomPassword();
        return {
            firstName: getRandomFirstName(),
            lastName: getRandomLastName(),
            email: generateEmail('normal'),
            password: password,
            confirmPassword: password,
            phoneNumber: ""
        };
    })(),

    invalidPhoneNumber: (() => {
        const password = generateRandomPassword();
        return {
            firstName: getRandomFirstName(),
            lastName: getRandomLastName(),
            email: generateEmail('normal'),
            password: password,
            confirmPassword: password,
            phoneNumber: "abcd1234567"  // intentionally invalid phone number
        };
    })(),

    emailWithoutAllDomain: (() => {
        const password = generateRandomPassword();
        return {
            firstName: getRandomFirstName(),
            lastName: getRandomLastName(),
            email: generateEmail('withoutAllDomain'),
            password: password,
            confirmPassword: password,
            phoneNumber: generateRandomPhoneNumber()
        };
    })(),

    emailWithoutTopLevelDomain: (() => {
        const password = generateRandomPassword();
        return {
            firstName: getRandomFirstName(),
            lastName: getRandomLastName(),
            email: generateEmail('withoutTopLevelDomain'),
            password: password,
            confirmPassword: password,
            phoneNumber: generateRandomPhoneNumber()
        };
    })(),

    emailDoubleAt: (() => {
        const password = generateRandomPassword();
        return {
            firstName: getRandomFirstName(),
            lastName: getRandomLastName(),
            email: generateEmail('doubleAt'),
            password: password,
            confirmPassword: password,
            phoneNumber: generateRandomPhoneNumber()
        };
    })(),

    emailDotBeforeAt: (() => {
        const password = generateRandomPassword();
        return {
            firstName: getRandomFirstName(),
            lastName: getRandomLastName(),
            email: generateEmail('dotBeforeAt'),
            password: password,
            confirmPassword: password,
            phoneNumber: generateRandomPhoneNumber()
        };
    })(),

    emailWithoutDomain: (() => {
        const password = generateRandomPassword();
        return {
            firstName: getRandomFirstName(),
            lastName: getRandomLastName(),
            email: generateEmail('withoutDomain'),
            password: password,
            confirmPassword: password,
            phoneNumber: generateRandomPhoneNumber()
        };
    })(),

    emailMissingUsername: (() => {
        const password = generateRandomPassword();
        return {
            firstName: getRandomFirstName(),
            lastName: getRandomLastName(),
            email: generateEmail('missingUsername'),
            password: password,
            confirmPassword: password,
            phoneNumber: generateRandomPhoneNumber()
        };
    })()
};

File: Hunt_Career_TestNG_Hybrid_Framework/src/main/java/com/huntcareer/qa/testdata/RegisterData.java
package com.huntcareer.qa.testdata;

import java.util.HashMap;
import java.util.Map;

import com.huntcareer.qa.utils.Utilities;

public class RegisterData {
	public static Map<String, String> validUser() {
		String password = Utilities.generateRandomPassword();
		Map<String, String>data = new  HashMap<>();
		data.put("firstName", Utilities.getFirstName());
		data.put("lastName", Utilities.getLastName());
		data.put("email", Utilities.getEmail("normal"));
		data.put("password", password);
		data.put("confirmPassword", password);
		data.put("phoneNumber", Utilities.generateRandomPhoneNumber());
		return data;
	}

	public static Map<String, String> duplicateUser() {
		Map<String, String> data = new  HashMap<>();
		data.put("firstName", "Pavan");
		data.put("lastName", "U");
		data.put("email", "upavan@mailinator.com");
		data.put("password", "Test@1234");
		data.put("confirmPassword", "Test@1234");
		data.put("phoneNumber", "1234567890");
		return data;
	}

	public static Map<String, String> noFirstName() {
		Map<String, String>data = new  HashMap<>();
		//data.put("firstName", Utilities.getFirstName());
		data.put("lastName", Utilities.getLastName());
		return data;
	}

	public static Map<String, String> noLastName() {
		Map<String, String>data = new  HashMap<>();
		data.put("firstName", Utilities.getFirstName());
		return data;
	}

	public static Map<String, String> noEmail() {
		String password = Utilities.generateRandomPassword();
		Map<String, String>data = new  HashMap<>();
		data.put("firstName", Utilities.getFirstName());
		data.put("lastName", Utilities.getLastName());
		//data.put("email", Utilities.getEmail("normal"));
		data.put("password", password);
		data.put("confirmPassword", password);
		return data;
	}

	public static Map<String, String> noPassword() {
		String password = Utilities.generateRandomPassword();
		Map<String, String>data = new  HashMap<>();
		data.put("firstName", Utilities.getFirstName());
		data.put("lastName", Utilities.getLastName());
		data.put("email", Utilities.getEmail("normal"));
		//data.put("password", password);
		data.put("confirmPassword", password);
		return data;
	}

	public static Map<String, String> passwordWithLessCharacters() {
		Map<String, String>data = new  HashMap<>();
		data.put("firstName", Utilities.getFirstName());
		data.put("lastName", Utilities.getLastName());
		data.put("email", Utilities.getEmail("normal"));
		data.put("password", "Test");
		data.put("confirmPassword", "Test");
		return data;
	}

	public static Map<String, String> noConfirmPassword() {
		String password = Utilities.generateRandomPassword();
		Map<String, String>data = new  HashMap<>();
		data.put("firstName", Utilities.getFirstName());
		data.put("lastName", Utilities.getLastName());
		data.put("email", Utilities.getEmail("normal"));
		data.put("password", password);
		//data.put("confirmPassword", password);
		return data;
	}

	public static Map<String, String> noPhoneNumber() {
		String password = Utilities.generateRandomPassword();
		Map<String, String>data = new  HashMap<>();
		data.put("firstName", Utilities.getFirstName());
		data.put("lastName", Utilities.getLastName());
		data.put("email", Utilities.getEmail("normal"));
		data.put("password", password);
		data.put("confirmPassword", password);
		data.put("phoneNumber", Utilities.generateRandomPhoneNumber());
		return data;
	}

	public static Map<String, String> invalidPhoneNumber() {
		String password = Utilities.generateRandomPassword();
		Map<String, String>data = new  HashMap<>();
		data.put("firstName", Utilities.getFirstName());
		data.put("lastName", Utilities.getLastName());
		data.put("email", Utilities.getEmail("normal"));
		data.put("password", password);
		data.put("confirmPassword", password);
		data.put("phoneNumber", "abcd123456");
		return data;
	}

	public static Map<String, String> emailWithoutAllDomain() {
		String password = Utilities.generateRandomPassword();
		Map<String, String>data = new  HashMap<>();
		data.put("firstName", Utilities.getFirstName());
		data.put("lastName", Utilities.getLastName());
		data.put("email", Utilities.getEmail("withoutAllDomain"));
		data.put("password", password);
		data.put("confirmPassword", password);
		data.put("phoneNumber", Utilities.generateRandomPhoneNumber());
		return data;
	}

	public static Map<String, String> emailwithoutTopLevelDomain() {
		String password = Utilities.generateRandomPassword();
		Map<String, String>data = new  HashMap<>();
		data.put("firstName", Utilities.getFirstName());
		data.put("lastName", Utilities.getLastName());
		data.put("email", Utilities.getEmail("withoutTopLevelDomain"));
		data.put("password", password);
		data.put("confirmPassword", password);
		data.put("phoneNumber", Utilities.generateRandomPhoneNumber());
		return data;
	}

	public static Map<String, String> emailwithdoubleAt() {
		String password = Utilities.generateRandomPassword();
		Map<String, String>data = new  HashMap<>();
		data.put("firstName", Utilities.getFirstName());
		data.put("lastName", Utilities.getLastName());
		data.put("email", Utilities.getEmail("doubleAt"));
		data.put("password", password);
		data.put("confirmPassword", password);
		data.put("phoneNumber", Utilities.generateRandomPhoneNumber());
		return data;
	}

	public static Map<String, String> emailWithdotBeforeAt() {
		String password = Utilities.generateRandomPassword();
		Map<String, String>data = new  HashMap<>();
		data.put("firstName", Utilities.getFirstName());
		data.put("lastName", Utilities.getLastName());
		data.put("email", Utilities.getEmail("dotBeforeAt"));
		data.put("password", password);
		data.put("confirmPassword", password);
		data.put("phoneNumber", Utilities.generateRandomPhoneNumber());
		return data;
	}

	public static Map<String, String> emailwithoutDomain() {
		String password = Utilities.generateRandomPassword();
		Map<String, String>data = new  HashMap<>();
		data.put("firstName", Utilities.getFirstName());
		data.put("lastName", Utilities.getLastName());
		data.put("email", Utilities.getEmail("withoutDomain"));
		data.put("password", password);
		data.put("confirmPassword", password);
		data.put("phoneNumber", Utilities.generateRandomPhoneNumber());
		return data;
	}

	public static Map<String, String> emailwithoutUserName() {
		String password = Utilities.generateRandomPassword();
		Map<String, String>data = new  HashMap<>();
		data.put("firstName", Utilities.getFirstName());
		data.put("lastName", Utilities.getLastName());
		data.put("email", Utilities.getEmail("withoutUserName"));
		data.put("password", password);
		data.put("confirmPassword", password);
		data.put("phoneNumber", Utilities.generateRandomPhoneNumber());
		return data;
	}
}
